nemo/collections/speechlm2/models/duplex_ear_tts.py

```diff
@@ -0,0 +1,1991 @@
+            "You must pass the config to DuplexEARTTS as a Python dict to support hyperparameter serialization "
+            f"in PTL checkpoints (we got: '{type(cfg)=}')."
+        )
+        super().__init__()
+        self.save_hyperparameters()
+        # convert dict to config
+        cfg = DictConfig(cfg)
+        self.trainer_config = cfg.get("trainer", None)
+        self.data_cfg = cfg.data
+        self.cfg = cfg.model
+        self.target_sample_rate = cfg.data.target_sample_rate
+        self.source_sample_rate = cfg.data.source_sample_rate
+        self.normalize_text = cfg.data.get("normalize_text", False)
+        self.model_16_precision_safe = None
+
+        self.validation_save_path = os.path.join(cfg.exp_manager.explicit_log_dir, "validation_logs")
+
+        # move back text channel by x, in inference it advance the text channel prediction by x frames
+        self.advance_text_channel_by = self.cfg.get("advance_text_channel_by", None)
+
+        # Load ForCausalLM
+        if self.cfg.tts_config.context_hidden_size is not None:
+            self.language_model = self._load_language_model(self.cfg)
+            self.embed_tokens = self._load_embed_tokens(self.cfg)
+            # delete llm because we use it only to get the  embbeding tokens
+            del self.language_model
+
+        # get codec run precision
+        self.audio_codec_run_dtype = getattr(torch, self.cfg.get("audio_codec_run_dtype", "float32"), torch.float32)
+
+        # instanciate eartts model and codec
+        self._load_tts_model(self.cfg)
+        self._codebook_size = self.tts_model.config.codebook_size
+
+        # compute source fps
+        self.source_fps = self.source_sample_rate / (
+            self.source_sample_rate * cfg.data.frame_length
+        )  # conver frame rate in fps
+        self.source_samples_per_frame = int(self.source_sample_rate // self.source_fps)
+
+        # get codec silence tokens
+        codec_silence_tokens = self.get_codec_silence_frame()
+        self.register_buffer("codec_silence_tokens", codec_silence_tokens)
+
+        # Load tokenizer
+        self.tokenizer = AutoTokenizer(
+            self.cfg.pretrained_lm_name, use_fast=True, trust_remote_code=True
+        )  # Note that we are using fast tokenizer
+
+        if 'Qwen2.5' in self.cfg.pretrained_lm_name:
+            # For Qwen, '<|im_start|>' is a common choice for a BOS token.
+            # You can check your tokenizer's vocabulary for the best candidate.
+            logging.warning("Tokenizer does not have a `bos_token`. Setting it to '<|im_start|>'.")
+            self.tokenizer.bos_token = '<|im_start|>'
+            self.tokenizer.eos_token = '<|im_end|>'
+
+        elif 'Nemotron' in self.cfg.pretrained_lm_name:
+            # ====== NEMOTRON-SPECIFIC HANDLING ======
+            self.tokenizer.bos_token = '<s>'
+            self.tokenizer.eos_token = '</s>'
+            self.tokenizer.pad_token = '<SPECIAL_12>'
```

Comment:
Ouch, let's not hardcode specific model names. Can we find a more general way that if qwen / if nemotron (I don't know this recipe well enough to suggest something specific atm)

Reply:
Makes sense, good catch. I replaced it by:

```
        self.tokenizer.bos_token = self.cfg.get("bos_token", '<s>')
        self.tokenizer.eos_token = self.cfg.get("eos_token", '</s>')
        self.tokenizer.pad_token = self.cfg.get("pad_token", '<SPECIAL_12>')
```

defaulting it to NanoV2, which is our current release candidate. Please let me know if you have any better suggestion.